packages/xstate-wallet/src/ui/close-ledger-and-withdraw.tsx

```diff
@@ -1,37 +1,120 @@
 import React from 'react';
-import {EventData} from 'xstate';
 import './wallet.scss';
-import {Button, Heading, Flex, Text} from 'rimble-ui';
-import {WorkflowState} from '../workflows/confirm';
+
+import {useService} from '@xstate/react';
+
+import {formatEther} from 'ethers/utils';
+import {Button, Heading, Flex, Text, Box, Link} from 'rimble-ui';
+import {getAmountsFromBudget} from './selectors';
+import {CloseLedgerAndWithdrawService} from '../workflows/close-ledger-and-withdraw';
+import {SiteBudget} from '../store';
 
 interface Props {
-  current: WorkflowState;
-  send: (event: any, payload?: EventData | undefined) => WorkflowState;
+  service: CloseLedgerAndWithdrawService;
 }
 
 export const CloseLedgerAndWithdraw = (props: Props) => {
-  const current = props.current;
-  const prompt = (
+  const [current, send] = useService(props.service);
+
+  const waitForUserApproval = ({waiting, budget}: {waiting: boolean; budget: SiteBudget}) => {
+    const {playerAmount, hubAmount} = getAmountsFromBudget(budget);
+    return (
+      <Flex alignItems="left" flexDirection="column">
+        <Heading textAlign="center" mb={0}>
+          Close your App Budget
+        </Heading>
+        <Heading textAlign="center" as="h4" mt={0} mb={2}>
+          {budget.domain}
+        </Heading>
+
+        <Text fontSize={1} pb={2}>
+          Close your budget with <strong>{budget.domain}</strong> and withdraw your funds?
+        </Text>
+
+        <Flex justifyContent="center" pb={2}>
+          <Box>
+            <Text>Send: {formatEther(playerAmount)} ETH</Text>
+            <Text>Receive: {formatEther(hubAmount)} ETH</Text>
+          </Box>
+        </Flex>
```

Comment:
"Current balances: You ..., Hub ..." ? Might not even need this table. All that really matters at this point is the amount that they stand to receive.